internal/models/factor.go

```diff
@@ -197,6 +210,14 @@ func (f *Factor) CreateChallenge(ipAddress string) *Challenge {
 	return challenge
 }
 
+func (f *Factor) CreateSMSChallenge(ipAddress string, otpCode string) *Challenge {
+	smsChallenge := f.CreateChallenge(ipAddress)
+	smsChallenge.OtpCode = otpCode
```

Comment:
I think we should `string(sha256.Sum256([]byte(otpCode)))` the `otpCode` before inserting it in the DB (and updating the check to do the same thing pre comparison).

It is an arguable improvement because it isn't a hard hash to crack, but that also isn't really what we are protecting against. It is more for the casual observer or point-in-time leak of the column where the OTP might still be valid

Reply:
Replied in comment above why I'm not a fan of hashing the OTP code in the database, if it's held encrypted.